streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPrint.java

```diff
@@ -62,13 +62,19 @@ public void init(ProcessorContext context) {
             if (valueSerde == null) {
                 this.valueSerde = context.valueSerde();
             }
-        }
 
-        @Override
-        public void process(final K key, final V value) {
-            final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
-            final V deValue = (V) maybeDeserialize(value, valueSerde.deserializer());
-            action.apply(deKey, deValue);
+            // no mapper provided, using a default one which takes care of checking byte[] for deserialization
+            if (mapper == null) {
+                mapper = new KeyValueMapper<K, V, String>() {
+                    @Override
+                    public String apply(K key, V value) {
+                        final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
```

Comment:
That is correct. In practice the stream is always `KStream<K, V>` and the serdes passed to print are also, `Serde<K>, Serde<V>`, so we'll never have the case where the stream is `KStream<byte[], byte[]>` and we use `Serde<Integer>, Serde<String>`, for example. So, the serde arguments as they stand are a bit useless. This was obviously an oversight when the functionality was first added. We'll rectify it in KIP-182